diesel/src/pg/types/ranges.rs

```diff
@@ -159,37 +159,37 @@ where
 }
 
 impl HasSqlType<Int4range> for Pg {
-    fn metadata(_: &PgMetadataLookup) -> PgTypeMetadata {
+    fn metadata(_: &Self::MetadataLookup) -> PgTypeMetadata {
```

Comment:
Just as sanity check: These changes should not be required, right?

Reply:
Yes, we would have to change this because the new type is `dyn PgMetadataLookup`. I got a compilation error when just putting `dyn PgMetadataLookup` in `HasSqlType<Array<T>>` because the recursive call required `'static` and it wasn't for some reason. Changing it to `Self::MetadataLookup` fixed the error some how. I thinking this is easier to refactor anyway so I changed all the `HasSqlType` implements for `Pg` to `Self::MetadataLookup`.

Reply:
That sounds for me like we should require that `PgMetadataLookup` is always `'static`.

Reply:
I think it is implied to be `'static` as part of the type of the trait (kind of like `Sized` if you don have `?Sized`) but not when in the function signature. I don't know why an error don't occur on the function signature though. I might post something on rust users forum to see if anyone understands what is going on.